akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -723,4 +730,73 @@ final class ProcessorModulePhase(materializer: PhasedFusingActorMaterializer) ex
   override def takePublisher(slot: Int, publisher: Publisher[Any]): Unit = publisher.subscribe(processor)
 
   override def onIslandReady(): Unit = ()
+}
+
+object TlsModuleIslandTag extends IslandTag
+
+final class TlsModulePhase(materializer: PhasedFusingActorMaterializer) extends PhaseIsland[TlsModulePhase.MaterializedInfo] {
+  import TlsModulePhase._
+
+  def name: String = "TlsModulePhase"
+
+  def materializeAtomic(mod: AtomicModule[Shape, Any], attributes: Attributes): (MaterializedInfo, Any) = {
+    val es = materializer.effectiveSettings(attributes)
+    println(mod)
+    val tls = mod.asInstanceOf[TlsModule]
```

Comment:
For the `TlsSpec` this fails for me with a `ClassCastException`. The module printed above is a

```
GraphStage(Map(<function1>)) [4313baad]
```

How can that be, @drewhk?